setup/mail-postfix.sh

```diff
@@ -210,6 +210,33 @@ tools/editconf.py /etc/postfix/main.cf \
 tools/editconf.py /etc/default/postgrey \
 	POSTGREY_OPTS=\"'--inet=127.0.0.1:10023 --delay=180'\"
 
+
+# We are going to setup a newer whitelist for postgrey, the version included in the distribution is old
+cat > /etc/cron.daily/mailinabox-postgrey-whitelist << EOF;
+#!/bin/bash
+
+# Mail-in-a-Box
+
+# check we have a postgrey_whitelist_clients file and that it is not older than 28 days
+if [ ! -f /etc/postgrey/whitelist_clients ] || find /etc/postgrey/whitelist_clients -mtime +28 ; then
+    # ok we need to update the file, so lets try to fetch it
+    if curl https://postgrey.schweikert.ch/pub/postgrey_whitelist_clients --output /tmp/postgrey_whitelist_clients -sS --fail > /dev/null 2>&1 ; then
+        # if fetching hasn't failed yet then check it is a plain text file
+        # curl manual states that --fail sometimes still produces output
+        # this final check will at least check the output is not html
+        # before moving it into place
+        if [ \$( file -b --mime-type /tmp/postgrey_whitelist_clients) == "text/plain" ]; then
```

Comment:
[shellcheck](https://www.shellcheck.net/) says this line is better as:

`if [ "$( file -b --mime-type /tmp/postgrey_whitelist_clients )" == "text/plain" ]; then`

Reply:
Ifve added that. The $ still needs escaping ofc.

Reply:
Ah yes because were are waiting for the EOF.

The "" are probably not really necessary as the mime-type shouldn't have a space in it but it will help protect the script from failure if some unknown value is returned I guess.